plugin/src/main/java/org/opensearch/sql/plugin/SQLPlugin.java

```diff
@@ -289,4 +297,16 @@ private DataSourceServiceImpl createDataSourceService() {
         dataSourceMetadataStorage,
         dataSourceUserAuthorizationHelper);
   }
+
+  @Override
+  public Collection<SystemIndexDescriptor> getSystemIndexDescriptors(Settings settings) {
+    List<SystemIndexDescriptor> systemIndexDescriptors = new ArrayList<>();
+    systemIndexDescriptors.add(
+        new SystemIndexDescriptor(
+            OpenSearchDataSourceMetadataStorage.DATASOURCE_INDEX_NAME, "SQL DataSources index"));
+    systemIndexDescriptors.add(
+        new SystemIndexDescriptor(
+            SPARK_REQUEST_BUFFER_INDEX_NAME + "*", "SQL Spark Request Buffer index pattern"));
```

Comment:
@noCharger does each datasource create its own request index?

Reply:
Seems like it is the case based on: https://github.com/opensearch-project/sql/blob/f241f3401fc4e73596727f3707e79c53a3290a1b/async-query-core/src/main/java/org/opensearch/sql/spark/execution/statestore/OpenSearchStateStoreUtil.java#L18

Reply:
Yes each datasource creates an index based on its name. Is it enough to define the pattern?

Reply:
What is the purpose of this registration?

Reply:
https://github.com/opensearch-project/security/issues/4535

Security currently maintains a list of system indices through an OpenSearch setting. This change will allow security to rely on a registry of these indices from the core instead of relying on a setting.

Reply:
Part of this greater effort: https://github.com/opensearch-project/security/issues/4439